matrix_client/api.py

```diff
@@ -590,7 +592,8 @@ def _send(self, method, path, content=None, query_params={}, headers={},
                 params=query_params,
                 data=content,
                 headers=headers,
-                verify=self.validate_cert
+                verify=self.validate_cert,
+                timeout=request_timeout
```

Comment:
Whatever error is raised by requests for a timeout needs to be handled and
reraised as a subclass of `MatrixError`. (maybe a new error class
`MatrixTimeoutError`?)

Reply:
Should we also handle the [ConnectTimeout](http://docs.python-requests.org/en/master/_modules/requests/exceptions/) next to the ReadTimeout here? With the same exception class? Maybe the ConnectTimeout should be handled with this [pull request](https://github.com/matrix-org/matrix-python-sdk/pull/123).